processors/filterprocessor/fields/filter.go

```diff
@@ -33,6 +34,28 @@ func (f Filter) SearchAndFilter(input data.BlockData) (data.BlockData, error) {
 
 	var newPayset []transactions.SignedTxnInBlock
 	switch f.Op {
+	case noneFieldOperation:
+		for _, txn := range input.Payset {
+
+			allFalse := true
+			for _, fs := range f.Searchers {
+				b, err := fs.search(txn)
+				if err != nil {
+					return data.BlockData{}, err
+				}
```

Comment:
This looks like the only bit that isn't covered by the unit tests. What are the failure conditions which would cause the searcher to return an error? Is this conditional something that should never happen or it could happen if you provide an invalid pattern?

Reply:
It just allows the searcher interface to propagate errors if the implementation needs to.  A quick example is the microalgos searcher which can return an error if the field is not a microalgo type (sort of belt and suspenders, but I think it is useful).